grakn-core/src/main/java/ai/grakn/graql/admin/ReasonerQuery.java

```diff
@@ -91,9 +91,10 @@
     /**
      * resolves the query
      * @param materialise materialisation flag
+     * @param explanation whether to provide explanation
      * @return stream of answers
      */
-    Stream<Answer> resolve(boolean materialise);
+    Stream<Answer> resolve(boolean materialise, boolean explanation);
```

Comment:
1. Your understanding is correct. Once materialised when asked for the explanation it will be explained on the basis of lookup retrieval.
2. Yes it does.